### PR TITLE
Slcb method transfer

### PR DESCRIPTION
**Pull request by SLCB-NN**

### CODE REVIEW

File: src/OptiHPLCHandler/empower_detector_module_method.py [hunk 1]
```diff
@@ -26,180 +26,296 @@ def xml_compatible(value: Union[str, bool]) -> str:
...
-            }
-        return channel_dict
+    def wavelengths(self) -> list[str]:
+        return [channel.wavelength for channel in self.channels]
+
+    @wavelengths.setter
+    def wavelengths(self, value: list[str]):
+        if not all(isinstance(v, (int, str)) for v in value):
+            raise ValueError("Wavelengths must be a list of strings or integers.")
+        self.channels = [TUVChannel(wavelength=wavelength) for wavelength in value]
```
Comment (SRFU-NN) [inline src/OptiHPLCHandler/empower_detector_module_method.py:130]: Does this work for int, or should it be 
```suggestion
        self.channels = [TUVChannel(wavelength=str(wavelength)) for wavelength in value]
```

File: tests/application_tests/test_method_converters.py [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+        self.assertTrue("CompositionD" in self.qsm_grad_ab[0])
+
+        # Run function
+        new_gradient_table = change_gradient_table(
+            self.qsm_grad_cd,
+            self.qsm_grad_ab,
+        )
+
+        # Check output
+        for grad1, grad2 in zip(self.qsm_grad_cd, new_gradient_table):
```
Comment (SRFU-NN) [inline tests/application_tests/test_method_converters.py:257]: Shouldn't we just check the whole gradient table?
```suggestion
        assertequal(self.qsm_grad_cd, new_gradient_table)
```

Ditto for any like-to-like substitution.

File: tests/test_detector.py [hunk 3]
```diff
@@ -53,20 +44,42 @@ def test_instantiation(self):
...
+
+    def test_set_channels_add(self):
+        self.assertEqual(self.method.channels[0].datamode, "SingleMode_1A")
+        self.assertEqual(self.method.channels[0].datarate, "SingleDataRate_20A")
+        self.assertEqual(self.method.channels[0].timeconstant, "0.1")
+        self.method.channels = [
+            TUVChannel(wavelength="400"),
+            TUVChannel(wavelength="500"),
+        ]
+        channels = self.method.channels
```
Comment (SRFU-NN) [inline tests/test_detector.py:65]: Should we also check the xml? That is what really matters, right? Either with tags, or just the values.

**Review by SRFU-NN — APPROVED** [review]
The only big suggestion is that we could check the content of the xml in method.current_method

File: src/OptiHPLCHandler/empower_detector_module_method.py [hunk 4]
```diff
@@ -25,184 +29,374 @@ def xml_compatible(value: Union[str, bool]) -> str:
...
+                f"{settings['channel']}"
+            )
+
+    @property
+    def wavelengths(self) -> list[str]:
+        return [channel.wavelength for channel in self.channels]
+
+    @wavelengths.setter
+    def wavelengths(self, value: list[str]):
+        if not all(isinstance(v, (int, str)) for v in value):
```
[the remark this answers is not in the document]
Comment (SLCB-NN) [reply src/OptiHPLCHandler/empower_detector_module_method.py:179]: Naaa, you cant have a float wavelength anyway

File: tests/application_tests/test_method_converters.py (hunk 2 above)
Comment (SLCB-NN) [reply tests/application_tests/test_method_converters.py:257]: Yeah no idea why i wasnt doing that...

Comment (SLCB-NN) [reply tests/application_tests/test_method_converters.py:257]: I guess because of the qsm and bsm comparison

File: tests/test_detector.py (hunk 3 above)
Comment (SLCB-NN) [reply tests/test_detector.py:65]: Ill do a basic, assertIsIn rather tahn looking through trees for now

**Pull request merged**